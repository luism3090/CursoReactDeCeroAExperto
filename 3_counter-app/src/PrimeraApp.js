import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

/* se debe colocar un div si querermos que el componente regrese varios elementos
mandamos la propiedad props y abajo colocamos props.variable  para imprirmi
NOTA : --> props ya no se se usa
*/

const PrimeraApp = (props) =>{

    const persona = {
        nombre:"Luis",
        edad:30
    }

    const hola = 'Hola mundo';

    //console.log(props);
    
    return (
        <div>
            <h2>{hola}</h2>
            <h3>{persona.nombre}</h3>
            <h3>{props.saludo}</h3>
            {/*<pre>{JSON.stringify(saludo,null,3)}</pre>*/}
            <p>Mi primera aplicación con React</p>
        </div>
    );

}


// si NO queremos colocar un div podemo colocar un Fragment y debemos importarlo
// esto hará que no aparesca ningun div en el html

const Componente1 = ({saludo2,saludo3,dato ="Hola mundo"}) => {
    
    return (
        <Fragment>
            <div>--------------------------------------------------</div>
            <h4>{saludo2}</h4>
            <h4>{saludo3}</h4>
            <h4>{dato}</h4>
            <p>Mi primera aplicación</p>
        </Fragment>
    );

}


// si NO queremos colocar un div podemo colocar un <> y </>
// esto hará que no aparesca ningun div en el html

const Componente2 = ({saludo,dato1,dato2,subtitulo}) => {
    
    return (
        <>
        <div>--------------------------------------------------</div>
            <h2>{saludo}</h2>
            <h2>{dato1}</h2>
            <h2>{dato2}</h2>
            <h3>{subtitulo}</h3>
            <p>Mi primera aplicación</p>
        </>
    );

}

Componente2.propTypes={
    saludo:PropTypes.string,
    dato1:PropTypes.number,
    dato2:PropTypes.string.isRequired
}

Componente2.defaultProps={
    subtitulo: 'Soy un subtitulo'
}


export{
    PrimeraApp as default,
    Componente1,
    Componente2
}
