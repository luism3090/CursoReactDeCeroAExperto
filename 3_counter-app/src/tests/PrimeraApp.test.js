import React from 'react';
import {shallow} from 'enzyme';
import '@testing-library/jest-dom';
import PrimeraApp,{Componente2} from '../PrimeraApp';

describe('Pruebas en <PrimeraApp>', () => {
    
    test('Debe mostrar el mensaje Hola, Soy Goku', () => {
        
        const saludo = 'Hola, soy Goku';
        const wrapper = shallow(<PrimeraApp saludo={saludo}/>);

        expect(wrapper).toMatchSnapshot();


    })
    
});


describe('Pruebas en <Componente2>', () => {
    
    test('Debe mostrar los saludos correctos', () => {
        
        const saludo = 'Hola, que tal';
        const dato1 = 123;
        const dato2 = 'Obligatorio';
        const subtitulo = 'Excelente subtitulo';
        const wrapper = shallow(<Componente2 
                                            saludo={saludo} 
                                            dato1={dato1} 
                                            dato2={dato2} 
                                            subtitulo={subtitulo}
                                            
                                />);

        // expect(wrapper).toMatchSnapshot();

        const textoH3 = wrapper.find('h3').text();

        console.log(textoH3);
        console.log(subtitulo);

        expect(textoH3).toBe(subtitulo);


    })
    
});
