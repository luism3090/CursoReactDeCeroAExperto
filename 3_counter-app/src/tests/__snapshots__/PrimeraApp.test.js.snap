// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp> Debe mostrar el mensaje Hola, Soy Goku 1`] = `
<div>
  <h2>
    Hola mundo
  </h2>
  <h3>
    Luis
  </h3>
  <h3>
    Hola, soy Goku
  </h3>
  <p>
    Mi primera aplicaci√≥n con React
  </p>
</div>
`;
